src/mmw/js/src/modeling/models.js

```diff
@@ -221,6 +221,54 @@ var ScenariosCollection = Backbone.Collection.extend({
             }
             model.set('active', active);
         });
+    },
+
+    duplicateScenario: function(cid) {
+        var source = this.get(cid),
+            sourceMods = source.get('modifications').models,
+            newModel = new ScenarioModel({
+                is_current_conditions: false,
+                name: this.makeNewScenarioName('Copy of ' + source.get('name'))
+            });
+
+        newModel.set('modifications', new ModificationsCollection(sourceMods, {
+            model: ModificationModel
+        }));
+
+        this.add(newModel);
+        this.setActiveScenario(newModel.cid);
+    },
+
+    // Generate a unique scenario name based off the baseName.
+    // Assumes a basic structure of "baseName X", where X is
+    // iterated as new scenarios with the same baseName are created.
+    // The first duplicate will not have an interated X in the name.
+    // The second duplicate will be "baseName 1".
+    makeNewScenarioName: function(baseName) {
+        var self = this,
+            regEx = new RegExp('^' + baseName + '( \\d|)+');
+
+        var numbers = _.without(this.map(function(model) {
+            var name = model.get('name'),
+                match = name.match(regEx);
+
+            if (match !== null) {
+                var matched = _.last(match);
+
+                if (matched !== '') {
+                    return parseInt(matched);
+                } else if (matched === '') {
+                    return 0;
+                }
+            }
+        }), undefined);
+
+        if (!_.isEmpty(numbers)) {
+            var max = _.max(numbers) + 1;
+            return baseName + ' ' + max;
+        } else {
+            return baseName;
+        }
```

Comment:
I think regexp is overkill for this. Could be made simpler with something like...

    var result = 'Copy of ' + name;
    if (!_.contains(existingNames, result)) return result;
    for (var i = 1; _.contains(existingNames, result + ' ' + i); i++);
    return result + ' ' + i;